libraries/botbuilder-dialogs-adaptive/src/languageGeneratorExtensions.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DialogManager } from 'botbuilder-dialogs';
+import { ResourceExplorer } from 'botbuilder-dialogs-declarative';
+import { LanguageGeneratorManager, ResourceMultiLanguageGenerator, TemplateEngineLanguageGenerator } from './generators';
+import { LanguageGenerator } from './languageGenerator';
+import { resourceExplorerKey } from './resourceExtensions';
+import { LanguagePolicy } from './languagePolicy';
+
+/**
+ * The key to get or set language generator from turn state.
+ */
+export const languageGeneratorKey = Symbol('LanguageGenerator');
+
+/**
+ * The key to get or set language generator manager from turn state.
+ */
+export const languageGeneratorManagerKey = Symbol('LanguageGeneratorManager');
+
+/**
+ * The key to get or set language policy from turn state.
+ */
+export const languagePolicyKey = Symbol('LanguagePolicy');
+
+/**
+ * Extension methods for language generator.
+ */
+export class LanguageGeneratorExtensions {
+    /**
+     * Register default LG file or a language generator as default language generator.
+     * @param dialogManager The dialog manager to add language generator to.
+     * @param lg LG resource id (default: main.lg) or language generator to be added.
+     * @returns dialog manager with language generator.
+     */
+    public static useLanguageGeneration(dialogManager: DialogManager, lg?: string | LanguageGenerator): DialogManager {
+        const resourceExplorer: ResourceExplorer = dialogManager.initialTurnState.get(resourceExplorerKey) || new ResourceExplorer();
+
+        let languageGenerator: LanguageGenerator;
+        if (!lg) {
+            lg = 'main.lg';
+        }
+        if (typeof(lg) === 'string') {
+            const resource = resourceExplorer.getResource(lg);
+            if (resource) {
+                languageGenerator = new ResourceMultiLanguageGenerator(lg);
+            } else {
+                languageGenerator = new TemplateEngineLanguageGenerator();
+            }
+        } else {
+            languageGenerator = lg;
+        }
```

Comment:
```suggestion
        let languageGenerator = lg || 'main.lg';

        if (typeof(lg) === 'string') {
            const resource = resourceExplorer.getResource(lg);
            if (resource) {
                languageGenerator = new ResourceMultiLanguageGenerator(lg);
            } else {
                languageGenerator = new TemplateEngineLanguageGenerator();
            }
        }
```

Reply:
If lg is undefined, languageGenerator cannot be initialized correctly.
How about use `if (typeof(languageGenerator) === 'string')` instead?

Reply:
good catch, I meant to replace those instances of `lg` but forgot to.